mbuild/conversion.py

```diff
@@ -0,0 +1,1390 @@
+    if box.angles is not None:
+        box_vector[3:6] = box.angles
+    else:
+        box_vector[3] = box_vector[4] = box_vector[5] = 90.0
+    for dim in range(3):
+        box_vector[dim] = box.lengths[dim] * 10
+    structure.box = box_vector
+    return structure
+
+def to_trajectory(compound, show_ports=False, chains=None,
+                  residues=None, box=None):
+    """Convert to an md.Trajectory and flatten the compound.
+
+    Parameters
+    ----------
+    show_ports : bool, optional, default=False
+        Include all port atoms when converting to trajectory.
+    chains : mb.Compound or list of mb.Compound
+        Chain types to add to the topology
+    residues : str of list of str
+        Labels of residues in the Compound. Residues are assigned by
+        checking against Compound.name.
+    box : mb.Box, optional, default=compound.boundingbox (with buffer)
+        Box information to be used when converting to a `Trajectory`.
+        If 'None', a bounding box is used with a 0.5nm buffer in each
+        dimension. to avoid overlapping atoms, unless `compound.periodicity`
+        is not None, in which case those values are used for the
+        box lengths.
+
+    Returns
+    -------
+    trajectory : md.Trajectory
+
+    See also
+    --------
+    _to_topology
+
+    """
+    md = import_('mdtraj')
+    atom_list = [particle for particle in compound.particles(show_ports)]
+
+    top = _to_topology(compound, atom_list, chains, residues)
+
+    # Coordinates.
+    xyz = np.ndarray(shape=(1, top.n_atoms, 3), dtype='float')
+    for idx, atom in enumerate(atom_list):
+        xyz[0, idx] = atom.pos
+
+    # Unitcell information.
+    unitcell_angles = [90.0, 90.0, 90.0]
+    if box is None:
+        unitcell_lengths = np.empty(3)
+        for dim, val in enumerate(compound.periodicity):
+            if val:
+                unitcell_lengths[dim] = val
+            else:
+                unitcell_lengths[dim] = compound.boundingbox.lengths[dim] + 0.5
+    else:
+        unitcell_lengths = box.lengths
+        unitcell_angles = box.angles
+
+    return md.Trajectory(xyz, top, unitcell_lengths=unitcell_lengths,
+                         unitcell_angles=unitcell_angles)
+
+def _to_topology(compound, atom_list, chains=None, residues=None):
+    """Create a mdtraj.Topology from a Compound.
+
+    Helper function for to_trajectory.
+    Parameters
+    ----------
+    atom_list : list of mb.Compound
+        Atoms to include in the topology
+    chains : mb.Compound or list of mb.Compound
+        Chain types to add to the topology
+    residues : str of list of str
+        Labels of residues in the Compound. Residues are assigned by
+        checking against Compound.name.
+
+    Returns
+    -------
+    top : mdtraj.Topology
+
+    See Also
+    --------
+    mdtraj.Topology : Details on the mdtraj Topology object
+
+    """
+    md = import_('mdtraj')
+    from mdtraj.core.topology import Topology
```

Comment:
Should this import be changed to `md.core.topology`??

Reply:
I tried but it failed the unit test. I guess the earlier call `md = import_('mdtraj')` is more to check if mdtraj is installed locally (or else it gonna raise an Error + instruction to install mdtraj)